packages/nimble/inst/tests/test-allocation.R

```diff
@@ -3,14 +3,18 @@
 library(nimble)
 source(system.file(file.path('tests', 'test_utils.R'), package = 'nimble'))
 
+RwarnLevel <- options('warn')$warn
```

Comment:
Long range dependencies like this can be difficult to maintain. Since you've already refactored to use a `test_that()` context elsewhere, this might be a good opportunity to introduce a testing helper that can use `on.exit` to clean up after itself.

As Hadley observed in his [testthat paper](https://journal.r-project.org/archive/2011-1/RJournal_2011-1_Wickham.pdf):

> When you use these actions in tests, you’ll need to clean up after yourself. Many other testing packages have set-up and teardown methods that are run automatically before and after each test. These are not so important with testthat because you can create objects outside of the tests and rely on R’s copy-onmodify semantics to keep them unchanged between test runs. To clean up other actions you can use regular R functions.